src/Symfony/Component/DependencyInjection/Compiler/AttributeAutoconfigurationPass.php

```diff
@@ -35,21 +82,81 @@ protected function processValue($value, bool $isRoot = false)
             || !$value->isAutoconfigured()
             || $value->isAbstract()
             || $value->hasTag('container.ignore_attributes')
-            || !($reflector = $this->container->getReflectionClass($value->getClass(), false))
+            || !($classReflector = $this->container->getReflectionClass($value->getClass(), false))
         ) {
             return parent::processValue($value, $isRoot);
         }
 
-        $autoconfiguredAttributes = $this->container->getAutoconfiguredAttributes();
         $instanceof = $value->getInstanceofConditionals();
-        $conditionals = $instanceof[$reflector->getName()] ?? new ChildDefinition('');
-        foreach ($reflector->getAttributes() as $attribute) {
-            if ($configurator = $autoconfiguredAttributes[$attribute->getName()] ?? null) {
-                $configurator($conditionals, $attribute->newInstance(), $reflector);
+        $conditionals = $instanceof[$classReflector->getName()] ?? new ChildDefinition('');
+
+        if ($this->classAttributeConfigurators) {
+            foreach ($classReflector->getAttributes() as $attribute) {
+                if ($configurator = $this->classAttributeConfigurators[$attribute->getName()] ?? null) {
+                    $configurator($conditionals, $attribute->newInstance(), $classReflector);
+                }
+            }
+        }
+
+        if ($this->methodAttributeConfigurators || $this->parameterAttributeConfigurators) {
+            $constructorReflector = $this->getConstructor($value, false);
+            if ($constructorReflector) {
+                foreach ($constructorReflector->getAttributes() as $attribute) {
```

Comment:
We should remove this loop: attributes on constructors don't make sense, ppl should use class-level attributes instead

Reply:
What about promoted properties?

Reply:
they will appear as both properties and params

Reply:
You're right!